rafiki/model/dataset.py

```diff
@@ -267,4 +282,38 @@ def _load(self, dataset_path):
 
         return (num_samples, num_classes, image_paths, image_classes, dataset_dir)
 
+class AudioFilesDataset(ModelDataset):
+    '''
+    Class that helps loading of dataset with type `AUDIO_FILES`
+
+    '''
+
+    def __init__(self, dataset_path, dataset_dir):
+        super().__init__(dataset_path)
+        self._dataset_dir = dataset_dir
+        (self.size, self.df) = self._load(self.path)
+
+    def _load(self, dataset_path):
+        dataset_dir = self._dataset_dir
+
+        dataset_zipfile = zipfile.ZipFile(dataset_path, 'r')
+        dataset_zipfile.extractall(path=dataset_dir.name)
+        dataset_zipfile.close()
+
+        # Read images.csv, and read image paths & classes
+        audios_csv_path = os.path.join(dataset_dir.name, 'audios.csv')
+
+
+        df = pandas.read_csv(audios_csv_path, encoding='utf-8', na_filter=False)
+        df.sort_values(by='wav_filesize', inplace=True)
+
+        # Convert to character index arrays
+        df['transcript'] = df['transcript']\
+            .apply(partial(text_to_char_array,
+                           alphabet=Alphabet('examples/datasets/speech_recognition/alphabet.txt')))
```

Comment:
That is definitely a good suggestion. I will consider moving the `alphabet.txt` file into the `.zip`.

Reply:
Problem occurred. After moving `alphabet.txt` into the `.zip`, the model has no access to `alphabet.txt` when doing prediction/inference alone.

Reply:
I think it is still best to place `alphabet.txt` in `examples/datasets/speech_recognition`, with a few advantages:
1. Avoid duplicates of `alphabet.txt` files in both training and testing dataset zips.
2. Can be used in inference jobs
3. As I have moved quoted chunk of code to `TfDeepSpeech.py`, it would not be used in the rafiki core code.
4. Yes you can customize `alphabet.txt` to Greek Alphabets by just editing the txt file to characters contained in the dataset transcripts.

Reply:
Have you considered saving an in-memory representation of `alphabet.txt` as part of the model's parameters (e.g. in `dump_parameters()` and `load_parameters(params)`)? You can think of the alphabet definition for the model as part of the *trained state* of the model, required for a full restoration of the trained model. This is similar to [`PyBiLstm.py`](https://github.com/nginyc/rafiki/blob/master/examples/models/pos_tagging/PyBiLstm.py), where associated metadata from the train dataset is also saved as part of the model's parameters. 

The main issue with the approach of reading from `examples/datasets/speech_recognition/` I see is that it introduces that dependency of the model code on the `examples/` code, and *that `alphabet.txt` in `examples/` would be static from the point of view of application developers*. 
Consider the scenario where an application developer submits a model training job (refer to [the users of Rafiki](https://nginyc.github.io/rafiki/docs/latest/src/dev/user-types.html)) to a running instance of Rafiki deployed on e.g. a cluster "NCRS". He would be unable to customize the alphabets associated with his dataset because he is unable to edit that `alphabet.txt` in the codebase of *that remote instance of Rafiki* because *it's not his local instance of Rafiki*. Essentially, if speech recognition models read from `examples/datasets/speech_recognition/alphabet.txt`, there can only be one file & one alphabets definition across all model training jobs - the one running Rafiki. 

Let me know if you think there's a mistake here.

Reply:
Understood your point. I haven't tried putting it in dumped params, but it is likely to work.

My only concern is that `Alphabet` may not be a 'MUST USE' for all speech recognition models, forcing users to provide alphabet info as a dataset standard may not be so appropriate.

An alternative solution for having only one remote instance of the file would be:
1. Model developer should create different models for different languages, each use a different `alphabet_config_path` to a different `alphabet.txt` file.
2. Application developers should only create train jobs using datasets correspond to the language.

Please let me know your opinion. Is the alternative solution or your proposal more appropriate?

Reply:
In context of your proposed alternative, does that mean that there'll be a separate task for each language e.g. `SPEECH_RECOGNITION_ENGLISH`, `SPEECH_RECOGNITION_FRENCH`? Currently, when *application developers* submit train jobs, they are only able to indirectly customize the Rafiki models that their datasets would be trained with by specifying the task (see https://nginyc.github.io/rafiki/docs/latest/src/user/quickstart-app-developers.html#creating-a-train-job). Conceptually, I feel that `SPEECH_RECOGNITION` is a single task, and the alphabet is a *property of the dataset* rather than the *property of the model template* (across different languages, the model code very likely won't need to change). In addition, to support `alphabet_config_path` we might need to create a new API on Rafiki and make changes to the core code.

With respect to this reasoning, I would say to opt for my proposal... Let me know if there's any drawbacks.

Reply:
Already put 'alphabet' into dumped params. See https://github.com/nginyc/rafiki/pull/132/commits/ee7c224aa8839e3956229739d9c64a52f6ebd368.